Apigee/Util/OrgConfig.php

```diff
@@ -190,17 +229,131 @@ public function __construct($org_name, $endpoint, $user, $pass, $options = array
                 }
             }
         }
-
-        if (array_key_exists('logger', $options) && $options['logger'] instanceof LoggerInterface) {
-            $this->logger = $options['logger'];
-        } else {
-            $this->logger = new NullLogger();
-        }
+        $this->http_options = $request_options;
         $this->user_mail = array_key_exists('user_mail', $options) ? $options['user_mail'] : null;
         $this->subscribers = $subscribers;
-        $this->http_options = $request_options;
-        $this->curl_options = $curl_options;
         $this->debug_callbacks = array_key_exists('debug_callbacks', $options) ? $options['debug_callbacks'] : array();
         $this->user_agent = array_key_exists('user_agent', $options) ? $options['user_agent'] : null;
+        if ($saml_error) {
+            $saml_error->orgConfig = $this;
+            throw $saml_error;
+        }
+    }
+
+    protected function getAccessTokenWithPasswordGrant($user, $pass, $saml_info)
+    {
+        $hash = md5(serialize(func_get_args()));
+        $cache_dir = sys_get_temp_dir() . '/edge-access-tokens';
+        $cache_file = "$cache_dir/$hash";
+        // See if we have an unexpired token cached for this user/pass combo.
+        if (file_exists($cache_file) && is_readable($cache_file)) {
+            $contents = json_decode($cache_file, true);
+            if (is_array($contents) && array_key_exists('token', $contents) && array_key_exists('expiry', $contents)) {
+                // If token is not expired, return it.
+                // Require fetching a new token 10 seconds before old one expires,
+                // to avoid unexpected failures mid-transaction.
+                if ($contents['expiry'] < time() - 10) {
+                    return $contents['token'];
+                }
+            }
+        }
+        // Create directory for token to be cached in, if it doesn't exist yet.
+        if (!is_dir($cache_dir)) {
+            @mkdir($cache_dir, 0777, true);
```

Comment:
If $cach_dir is not created, the error will never be displayed.  I don't think this is an issue since the tmp dir should always be available.  Is it OK for it to be so open in permissions?